### PR TITLE
Added new initil assumptions with boostings

### LINKED ISSUE #1341: enh: Design effective initial assumption
The initial assumtions should be actualized using most actual exemples from other AutoML (moslty boosting-based).

The effect should be confirmed on AutoMLBenchmark.

### PR DESCRIPTION
**Pull request by dmitryglhf**
This is a 🔨 code refactoring.

## Summary

- **New Initial Assumptions:** Updated initial assumptions by adding boosting-based solutions (CatBoost, XGBoost, LightGBM).
  
  Comparison table between old and new assumptions (validated on automlbenchmark small dataset 1h8c):

  |    | Metric (mean)            |   main (Random Forest first) |   new (GBM joined with Linear model) |
  |---:|:------------------|-------------:|-------------------:|
  |  0 | auc               |     0.869263 |           0.879746 |
  |  1 | acc               |     0.84667  |           0.852339 |
  |  2 | balacc            |     0.805336 |           0.822745 |
  |  3 | logloss           |     0.449189 |           0.377827 |
  |  4 | training_duration, s |   242.554    |        251.445     |

## Context

  Closes #1341

### CODE REVIEW

**Comment by dmitryglhf** [comment]
@nicl-nno Среди трех начальных приближений стоит оставить все или же только то, которое улучшает метрики лучше  остальных?

**Comment by nicl-nno** [comment]
>Среди трех начальных приближений стоит оставить все или же только то, которое улучшает метрики больше остальных?

Зависит от того, меняется ли лидер при смене группы датасетов. Если везде +-один вариант доминирует - то можно его иоставить.

**Comment by dmitryglhf** [comment]
Looks like FEDOT mostly choosing and modifying`gbm_linear` assumption.

<details>

<summary>Details</summary>

Binary classification:
```python
binary_class_datasets = [
    'blood-transf.arff.csv', 'christine.arff.csv', 'jasmine.arff.csv',
    'phoneme.arff.csv', 'sylvine.arff.csv',
]
```

image

Multiclass classification:
```python
multi_class_datasets = [
    'car.arff.csv', 'cnae-9.arff.csv', 'dilbert.arff.csv', 'fabert.arff.csv',
    'mfeat-factors.arff.csv', 'segment.arff.csv', 'vehicle.arff.csv'
]
```

image


Regression:
```python
regression_datasets = [
    'analcatdata_negotiation.arff.csv', 'bodyfat.arff.csv', 'cleveland.arff.csv', 
    'cloud.arff.csv', 'kin8nm.arff.csv', 'liver-disorders.arff.csv'
]
```

image


Logs:
[class_compare_datasets.log](https://github.com/user-attachments/files/18628882/class_compare_datasets.log)
[regr_compare_datasets.log](https://github.com/user-attachments/files/18628881/regr_compare_datasets.log)

</details>

**Comment by nicl-nno** [comment]
>Looks like FEDOT mostly choosing and modifyinggbm_linear assumption

А это хорошо или плохо?

**Comment by dmitryglhf** [comment]
> > Looks like FEDOT mostly choosing and modifyinggbm_linear assumption
> 
> А это хорошо или плохо?

Это промежуточное сообщение, чтобы не потерять результаты, хотел по нему задать вопрос. 
В нем сравнение какое начальное приближение доминирует. В задачах классификации преимущественно выбирается за основу или как итоговое решение первый пайплайн `gbm_linear`.
В случае с регрессией похоже за основу берется тот же пайплайн, но при этом он сильно меняется. Может быть стоит оставить текущее приближение `rfr`?

> Regression:
> 
> ```python
> regression_datasets = [
>     'analcatdata_negotiation.arff.csv', 'bodyfat.arff.csv', 'cleveland.arff.csv', 
>     'cloud.arff.csv', 'kin8nm.arff.csv', 'liver-disorders.arff.csv'
> ]
> ```
> 
> image
> 
> Logs: [regr_compare_datasets.log](https://github.com/user-attachments/files/18628881/regr_compare_datasets.log)

**Review by nicl-nno — APPROVED** [review]
